frontend/app/(tabs)/home.tsx

```diff
@@ -72,14 +62,42 @@ const Index = () => {
     } finally {
       setIsLoading(false);
     }
-  }, [page, isLoading]);
+  }, [page, isLoading, searchText]);
+
+  const handleSearch = (text: string) => {
+    setSearchText(text);
+
+    if (searchTimeout.current) {
+      clearTimeout(searchTimeout.current);
+    }
```

Comment:
Oh I see what the timeout is for, is this to add a delay to the sending the query? I think we can solve this in a more basic way by debouncing the input, I think that's how we use to solve it in the react course for the jobs, can you double check there please?